### ISSUE TITLE
Do not create an empty commit when generating report

### ISSUE THREAD

**Issue by timtebeek** [spring-projects-experimental/spring-boot-migrator#527]
**Describe the bug**
When I create a report to inform myself of the steps needed for a migration, an empty commit is created and added to the git history. As a user this is quite surprising; I would not expect a report generation that does not yet make any code changes to add anything to my git history. Perhaps that's intentional, but I'd thought to mildly challenge that here, as it makes the report generation more widely applicable if it does not create any git changes (yet).

**To Reproduce**
1. `git clone git@github.com:spring-projects-experimental/spring-boot-migrator.git`, currently at https://github.com/spring-projects-experimental/spring-boot-migrator/commit/21fce4d2ee965de11f323665ca9492fed83e615f
2. `cd spring-boot-migrator`
3. `./mvnw install -DskipTests`
4. `java -jar applications/spring-shell/target/spring-boot-migrator.jar`
5. `scan ../any-other-project`
6. `apply --recipeName boot-2.7-3.0-upgrade-report2`

**Expected behavior**
Only a generated report; no new commits added to the repository.

Instead I see a new commit:

> SBM: applied recipe 'boot-2.7-3.0-upgrade-report2'

**Desktop (please complete the following information):**
 - OS: Ubuntu 2022.04
 - Version https://github.com/spring-projects-experimental/spring-boot-migrator/commit/21fce4d2ee965de11f323665ca9492fed83e615f

**Comment by fabapp2** [comment]
Hm... I agree that an empty commit is useless and should be fixed. Not sure if I agree that no commit at all should be created. But I'm Interested in your reasoning.

**Comment by timtebeek** [comment]
In my view report generation is purely an informative step, to better judge any follow up code charging steps to take after that. The reports do not yet make code changes, and it's unlike that you'd want the report to be added to the repository. Hence why I would not yet expect a commit on the active branch, much less an empty commit at that.

Perhaps there'd be some value if the report commit is on a separate branch. But when users have the main branch checked out and generate a report, they could be surprised to find a new empty commit on the main branch, which typically can not be pushed directly for protected branches.

**Comment by fabapp2** [comment]
Hi @timtebeek 
we reworked the Spring Boot Report from ground up. The shell is history for report generation (recipes and report generation should still work though) and will be (see branch `webapp-demo`) replaced with the HTML report that allows applying recipes from the report. the empty commit problem has been fixed in this branch. Using the Shell will currently still create a commit but when using the new report capability there are no commits for the generated report.  
I like the new approach a lot and could see this replacing the shell completely over time.
